### PR TITLE
[ADD] reporting-context

### PR DESCRIPTION
**Pull request by jarroyomorales**
This module adds a context variable to reports. A possible use for this
context could be hiding some fields or many other configuration options.

To configure this module, you need to:

* Enter Odoo in debug mode.
* Go to Settings -> Parameters -> System Parameters. On the system
  parameters list, look for report.default.context, which is a Python
  dictionary variable where you can add a context that will be common for
  all reports.
* To add a specific context to a report, you should go to Settings ->
  Reporting -> Reporting and look for the report you want to edit on the
  list. You will see that now they contain a new field called Context Value
  , where you will be able to add all the desired context parameters.

### CODE REVIEW

**Comment by jarroyomorales** [comment]
This is the addon that we were talking about on OCA/l10n-spain#1014

File: report_context/tests/__init__.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl-3).
+
+from . import test_report_context
```
Comment (etobella) [inline report_context/tests/__init__.py]: Missing new line

File: report_context/tests/__init__.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl-3).
```
Comment (etobella) [inline report_context/tests/__init__.py]: It is not necessary if it is in the __init__

**Review by etobella — APPROVED** [review]
Just a few comments. But it looks great for me

**Review by JordiBForgeFlow — APPROVED** [review]
:+1: Tested and reviewed code. Works as expected. Notice that in order to work effectively in some cases it need this fix in Odoo https://github.com/odoo/odoo/pull/31150

**Comment by JordiBForgeFlow** [comment]
@jarroyomorales can you squash your commits?

**Comment by pedrobaeza** [comment]
Why not adding the context in the report object in a similar way that action windows have a context?

**Comment by etobella** [comment]
Well, it can be edited like 
```xml
<record id="model_name.report_id" model="ir.actions.report">
    <field name="context">YOUR CONTEXT HERE</field>
</record>
```
It is hard to edit the standard function where it reads it, as it is on `odoo.tools`.

**Comment by pedrobaeza** [comment]
Yeah, that's the idea. It's for having the information in the same place, instead of using some kind of black magic in names for fetching from system parameters the context.

**Comment by etobella** [comment]
This module allows to use both systems. A generic parameter and specific report parameters. Maybe it should be explained on the readme.

**Comment by pedrobaeza** [comment]
I only saw the part of the system parameter, but for me the most important one should be the other. Even more, I would remove the system parameter feature, as it's redundant, can produce permission problems, and you already have the other mechanism.

**Comment by etobella** [comment]
Well, the system parameter might be important as sometimes you have a standard behaviour that should be found on all reports. But it is not the interesting part of the module.

**Comment by pedrobaeza** [comment]
OK, I see. Maybe giving more importance to the other part in the README avoids confusions like mine.

**Comment by pedrobaeza** [comment]
But I admit the problem was to not reading full README. If you don't want to spend more time, I can merge it.

**Comment by etobella** [comment]
@jarroyomorales can you improve the README and rebase all in order to merge it, please?

**Comment by jarroyomorales** [comment]
Done, I changed a little bit the readme and squashed the commits!

**Comment by JordiBForgeFlow** [comment]
Yes, it is better understood now!

**Pull request merged**